src/data/docs/abilities.md

```diff
@@ -334,6 +334,88 @@ In implementing this, you'll have a handler that matches on a `Request {Ask a, S
 
 Once you've written `pipe`, try writing `Stream.map`, `Stream.filter`, and `Stream.take` using `pipe`.
 
+### Handling abilities into IO
+
+In the introduction, the `greet` function was defined in terms of the `IO` ability — a special ability provided by the Unison runtime to interact with the environment (i.e., perform file or socket I/O, read from the local system clock, raise errors, etc.). In real programs, however, developers should generally follow the _principle of least power_ by describing such interactions in terms of more limited abilities. Carefully constraining the operations that a function may access makes our programs easier to reason about and easier to test. Limited abilities can then be interpreted in terms of `IO` at the top level of a program.
+
+Let's implement a Hello World program in terms of a user-defined `Console` ability that we'll handle into `IO`.
+
+```ability
+Console where
+  readLine  : Text
+  printLine : Text -> ()
```

Comment:
Looks like the transcript runner reformatted these lines and broke the markdown (`ability` will be probably be interpreted as a language value for the code block). Same thing happened elsewhere...